### PR TITLE
fix: Mappings View for Activity

### PR DESCRIPTION
**Pull request by ruuushhh**
### Description
Mappings View for Activity
image

## Clickup
[Handle Mappings for Activity Category](https://app.clickup.com/t/86cwkgp4t)

### CODE REVIEW

File: fyle_accounting_mappings/views.py [hunk 1]
```diff
@@ -79,6 +79,15 @@ def get_queryset(self):
...
+            if source_type == 'CATEGORY':
+                activity_attribute = ExpenseAttribute.objects.filter(
+                    attribute_type='CATEGORY', value='Activity', workspace_id=self.kwargs['workspace_id'], active=True).first()
+                activity_mapping = Mapping.objects.filter(
+                    source_type='CATEGORY', source__value='Activity', workspace_id=self.kwargs['workspace_id']).first()
+                
+                if activity_attribute and not activity_mapping:
+                    params = params & ~Q(source__value='Activity')
+
             mappings = Mapping.objects.filter(**params)
```
Comment (ashwin1111) [inline fyle_accounting_mappings/views.py:91]: can we just do `mappings = Mapping.objects.filter(**params, ~Q(source__value='Activity'))`?

File: fyle_accounting_mappings/views.py [hunk 2]
```diff
@@ -317,42 +326,57 @@ def get_queryset(self):
...
+            attribute_type='CATEGORY', value='Activity', workspace_id=self.kwargs['workspace_id'], active=True).first()
+
+        # Adjust the filters if 'Activity' attribute exists but not mapped
+        if activity_attribute and not activity_mapping:
+            base_filters &= ~Q(value='Activity')
 
+        # Handle the mapped parameter
         param = None
-        if mapped:
+        if mapped is True:
```
Comment (ashwin1111) [inline fyle_accounting_mappings/views.py:369]: frontend query params will be parsed as string 'true', let the condition be as it is

File: fyle_accounting_mappings/views.py [hunk 3]
```diff
@@ -317,42 +326,57 @@ def get_queryset(self):
...
             param = ~Q(categorymapping__source_category_id__in=source_categories)
-        else:
-            return ExpenseAttribute.objects.filter(Q(**filters)).order_by('value')
-        final_filter = Q(**filters)
+
+        # Combine the base filters with the param (if any)
+        final_filter = base_filters
         if param:
-            final_filter = final_filter & param
+            final_filter &= param
```
Comment (ashwin1111) [inline fyle_accounting_mappings/views.py:377]: can discuss in tomo's call

File: fyle_accounting_mappings/views.py (hunk 2 above)
Comment (ruuushhh) [reply fyle_accounting_mappings/views.py:369]: we are setting mapped above

File: fyle_accounting_mappings/views.py (hunk 3 above)
Comment (ruuushhh) [reply fyle_accounting_mappings/views.py:377]: discussed 
Added the else condition

File: fyle_accounting_mappings/views.py (hunk 1 above)
Comment (ruuushhh) [reply fyle_accounting_mappings/views.py:91]: this way is better no
also works fine

Comment (ashwin1111) [reply fyle_accounting_mappings/views.py]: this causes unnecessary 2 queries which is checking expense attributes and mappings, we can directly do <> activity

File: fyle_accounting_mappings/views.py [hunk 4]
```diff
@@ -317,42 +332,59 @@ def get_queryset(self):
...
 
+        # Get the 'Activity' mapping and attribute
+        activity_mapping = CategoryMapping.objects.filter(
+            source_category__value='Activity', workspace_id=self.kwargs['workspace_id']).first()
+        activity_attribute = ExpenseAttribute.objects.filter(
+            attribute_type='CATEGORY', value='Activity', workspace_id=self.kwargs['workspace_id'], active=True).first()
+
+        # Adjust the filters if 'Activity' attribute exists but not mapped
+        if activity_attribute and not activity_mapping:
+            base_filters &= ~Q(value='Activity')
```
Comment (ashwin1111) [inline fyle_accounting_mappings/views.py:371]: same comment as above

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**